src/producer.rs

```diff
@@ -0,0 +1,84 @@
+use rust_arroyo::backends::kafka::config::KafkaConfig;
+use rust_arroyo::backends::kafka::producer::KafkaProducer;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+
+use rust_arroyo::types::{Topic, TopicOrPartition};
+
+use crate::types::{
+    CheckResult,
+};
+use rust_arroyo::backends::{Producer, ProducerError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExtractCodeError {
+    #[error(transparent)]
+    SerializationError(#[from] serde_json::Error),
+    #[error(transparent)]
+    ProducerError(#[from] ProducerError),
+}
+
+pub async fn produce_checker_result(
+    result: CheckResult,
+    topic: Topic,
+    config: KafkaConfig,
+) -> Result<(), ExtractCodeError> {
+    let producer = KafkaProducer::new(config.clone());
+    let topic = TopicOrPartition::Topic(topic);
+    // TODO: Note that this will rarely fail directly here - mostly will only happen if the queue
+    // is full. We need to have a callback that tells us whether the other produces succeed so we
+    // know whether to mark the check as successful. We never want to just silently drop a result -
+    // we'd much prefer to retry it later and produce a miss, so that we at least know that we missed
+    // and don't have holes.
+    Ok(producer.produce(
+        &topic,
+        KafkaPayload::new(None, None, Some(serde_json::to_vec(&result)?)),
+    )?)
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::NaiveDateTime;
+    use chrono::offset::Local;
```

Comment:
Unused I think

Reply:
Oh yeah, that's some test junk I should remove